DNN Platform/Library/Common/Globals.cs

```diff
@@ -1,3842 +1,3935 @@
+                {
+                    // Return value
+                    returnValue = validClassLookupDictionary[inputValue];
+                    idFound = true;
+                }
+            }
+
+            if (!idFound)
+            {
+                using (ISharedCollectionLock writeLock = validClassLookupDictionary.GetWriteLock())
+                {
+                    if (!validClassLookupDictionary.ContainsKey(inputValue))
+                    {
+                        // Create Valid Class
+                        // letters ([a-zA-Z]), digits ([0-9]), hyphens ("-") and underscores ("_") are valid in class values
+                        // Remove all characters that aren't in the list
+                        returnValue = InvalidCharacters.Replace(inputValue, string.Empty);
+
+                        // If we're asked to validate the first character...
+                        if (validateFirstChar)
+                        {
+                            // classes should begin with a letter ([A-Za-z])'
+                            // prepend a starting non-letter character with an A
+                            if (InvalidCharacters.IsMatch(returnValue))
+                            {
+                                returnValue = "A" + returnValue;
+                            }
+                        }
+
+                        // put in Dictionary
+                        validClassLookupDictionary[inputValue] = returnValue;
+                    }
+                }
+            }
+
+            // Return Value
+            return returnValue;
+        }
+
+        /// <summary>
+        ///   Creates the valid ID.
+        /// </summary>
+        /// <param name = "inputValue">The input value.</param>
+        /// <returns>String with a valid ID.</returns>
+        public static string CreateValidID(string inputValue)
+        {
+            string returnValue = Null.NullString;
+
+            // Regex is expensive so we will cache the results in a lookup table
+            var validIDLookupDictionary = CBO.GetCachedObject<SharedDictionary<string, string>>(
+                new CacheItemArgs("ValidIDLookup", 200, CacheItemPriority.NotRemovable),
+                (CacheItemArgs cacheItemArgs) => new SharedDictionary<string, string>());
+
+            bool idFound = Null.NullBoolean;
+            using (ISharedCollectionLock readLock = validIDLookupDictionary.GetReadLock())
+            {
+                if (validIDLookupDictionary.ContainsKey(inputValue))
+                {
+                    // Return value
+                    returnValue = validIDLookupDictionary[inputValue];
+                    idFound = true;
+                }
+            }
+
+            if (!idFound)
+            {
+                using (ISharedCollectionLock writeLock = validIDLookupDictionary.GetWriteLock())
+                {
+                    if (!validIDLookupDictionary.Contain
```

Comment:
This TODO has been handled, right?
```suggestion
```

Reply:
Look like the code comment is out of place but if it was the only other TODO, it's resolved in https://github.com/dnnsoftware/Dnn.Platform/pull/4215/commits/995133d2e7a70acee2b9ecbc27bdf86171d70119